dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImplTest.java

```diff
@@ -330,6 +332,115 @@ public void testIsCheckInSafeWithoutRelationshipsShouldReturnTrue() {
+     * ExpectedResult: Should throw a DotSecurityException
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test(expected = DotSecurityException.class)
+    public void whenTryToLockShouldThrowDotSecurityException() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+
+        final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
+        final Contentlet contentletSaved = esContentletAPI.checkin(contentlet, APILocator.systemUser(), false);
+        esContentletAPI.lock(contentletSaved, user, false);
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#lock(Contentlet, User, boolean)}
+     * Given Scenario: A user with {@link PermissionLevel#EDIT} permission try to lock a contentlet
+     * ExpectedResult: Should work
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test
+    public void whenTryToLockShouldWork() throws DotSecurityException, DotDataException {
+        final Role role = new RoleDataGen().nextPersisted();
+        final User user = new UserDataGen().roles(role).nextPersisted();
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet = new ContentletDataGen(contentType.id()).next();
+
+        final ESContentletAPIImpl esContentletAPI = new ESContentletAPIImpl();
+        final Contentlet contentletSaved = esContentletAPI.checkin(contentlet, APILocator.systemUser(), false);
+
+        addPermission(role, contentType, PermissionLevel.WRITE);
+        addPermission(role, contentletSaved, PermissionLevel.WRITE);
+
+        esContentletAPI.lock(contentletSaved, user, false);
+
+        checkLock(user, contentletSaved);
+    }
+
+    private void checkLock(final User user, final Contentlet contentletSaved) throws DotDataException {
+        final ContentletVersionInfo info = APILocator.getVersionableAPI().
+                getContentletVersionInfo(contentletSaved.getIdentifier(), contentletSaved.getLanguageId());
+
+        assertNotNull(info.getLockedBy());
+        assertNotNull(info.getLockedOn());
+        assertEquals(user.getUserId(), info.getLockedBy());
+    }
+
+    /**
+     * Method to test: {@link ESContentletAPIImpl#lock(Contentlet, User, boolean)}
+     * Given Scenario: The contentlet's owner without permission to EDIT try to lock the contebtlet
+     * ExpectedResult: Should work
+     *
+     * @throws DotSecurityException
+     * @throws DotDataException
+     */
+    @Test()
+    public void whenOwnerTryToLockShouldWork() throws DotSecurityException, DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5365897)